### PR TITLE
Patch abseil-cpp to ignore deprecated errors in new Xcode.

### PR DESCRIPTION
**Pull request by jonsimantov**
The current version of abseil-cpp (used by grpc) uses some deprecated builtins, which cause warnings that make the build fail on Xcode 15 (due to warnings-as-errors).

Until grpc is updated, this PR will work around the issue by adding a patch to abseil-cpp to turn off that warning for that file.

### CODE REVIEW

File: cmake/external/abseil-cpp.patch.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+diff --git a/absl/meta/type_traits.h b/absl/meta/type_traits.h
+index d886cb30..de0ebda2 100644
+--- a/absl/meta/type_traits.h
++++ b/absl/meta/type_traits.h
+@@ -35,6 +35,12 @@
+ #ifndef ABSL_META_TYPE_TRAITS_H_
+ #define ABSL_META_TYPE_TRAITS_H_
+ 
++// Added by firestore-ios-sdk/cmake/external/abseil-cpp.patch.txt.
++#if __APPLE__
```
Comment (dconeybe) [inline cmake/external/abseil-cpp.patch.txt:10]: Drive-by comment: I'd recommend doing an ifdef for clang rather than apple:

```
#if __clang__
```

https://clang.llvm.org/docs/LanguageExtensions.html

Comment (jonsimantov) [reply cmake/external/abseil-cpp.patch.txt:10]: OK, that'll definitely catch Xcode though?

Comment (dconeybe) [reply cmake/external/abseil-cpp.patch.txt]: Yes, it absolutely should.

File: cmake/external/abseil-cpp.patch.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+diff --git a/absl/meta/type_traits.h b/absl/meta/type_traits.h
+index d886cb30..84c01e19 100644
+--- a/absl/meta/type_traits.h
++++ b/absl/meta/type_traits.h
+@@ -35,6 +35,12 @@
+ #ifndef ABSL_META_TYPE_TRAITS_H_
+ #define ABSL_META_TYPE_TRAITS_H_
+ 
++// Added by firestore-ios-sdk/cmake/external/abseil-cpp.patch.txt.
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply cmake/external/abseil-cpp.patch.txt:9]: Oh it should say firebase-ios-sdk anyway - that's where the patch is located. Is that ok?

[the remark this answers is not in the document]
Comment (dconeybe) [reply cmake/external/abseil-cpp.patch.txt:9]: I'm confused. Isn't this the patch file right here, in this file, that the comment should be referring to?

[the remark this answers is not in the document]
Comment (dconeybe) [reply cmake/external/abseil-cpp.patch.txt:9]: Did you mean firebase-**cpp**-sdk? (i.e. "cpp" instead of "ios")?

[the remark this answers is not in the document]
Comment (jonsimantov) [reply cmake/external/abseil-cpp.patch.txt:9]: Yeah, this patch file right here that is in the firebase-ios-sdk

[the remark this answers is not in the document]
Comment (dconeybe) [reply cmake/external/abseil-cpp.patch.txt]: :man_facepalming: I thought we were in the firebase-cpp-sdk. Nevermind!

**Review by dconeybe — APPROVED** [review]

**Comment by Vyazovoy** [comment]
Hi, @jonsimantov! Doesn't #12538 means that your changes should be reverted?
